src/openms/include/OpenMS/ANALYSIS/MAPMATCHING/QTClusterFinder.h

```diff
@@ -112,10 +113,24 @@ namespace OpenMS
 
     /// Map to store which grid features are next to which clusters
     typedef OpenMSBoost::unordered_map<
-              OpenMS::GridFeature*, std::vector<QTCluster*> > ElementMapping;
+              OpenMS::GridFeature*, std::unordered_set<Size> > ElementMapping;
+
+    /// Handles for the clusters in the index
+    typedef boost::heap::fibonacci_heap<QTCluster>::handle_type Handle;
+
+    /// Heap to efficiently find the best clusters
+    typedef boost::heap::fibonacci_heap<QTCluster> Heap;
+
+    /// Map of neighbors of a cluster
+    typedef std::pair<double, GridFeature*> NeighborPairType;
+    typedef OpenMSBoost::unordered_map<Size, NeighborPairType> NeighborMap;
+    
+    /// Map of elements of a cluster (neighbors + center)
+    typedef NeighborMap ClusterElementsMap;
```

Comment:
a typedef of a typedef?

Reply:
It is the same internal type used in different situations. The problem is that sometimes we need just the neighbors and sometimes also the cluster center (see comment line 128). If the name would still be NeighborMap it would be wrong. I will try to come up with a name that fits both situations and make it a single typedef.